crates/ethereum/payload/src/lib.rs

```diff
@@ -244,6 +239,26 @@ where
 
         Ok(EthBuiltPayload::new(attributes.payload_id(), sealed_block, U256::ZERO))
     }
+
+    fn cfg_and_block_env(
+        &self,
+        chain_spec: &ChainSpec,
+        parent: &Header,
+    ) -> (CfgEnvWithHandlerCfg, BlockEnv) {
+        let mut cfg_env = CfgEnvWithHandlerCfg::new_with_spec_id(CfgEnv::default(), SpecId::LATEST);
```

Comment:
using the latest spec id is not necessarily correct, as some forks may not be activated yet. please bring back the previous activation logic

Reply:
ignore - we override this later on. not obvious!